crates/cryptography/src/witness.rs

```diff
@@ -0,0 +1,135 @@
+    /// list of eigenda blob
+    pub eigenda_blobs: Vec<Bytes>,
+    /// list of commitments
+    pub commitments: Vec<Bytes>,
+    /// list of proofs
+    pub proofs: Vec<Bytes>,
+}
+
+/// impl EigenDABlobWitness
+impl EigenDABlobWitness {
+    /// Create a default struct
+    pub fn new() -> Self {
+        EigenDABlobWitness {
+            eigenda_blobs: Vec::new(),
+            commitments: Vec::new(),
+            proofs: Vec::new(),
+        }
+    }
+
+    /// This function computes a witness for a eigenDA blob
+    /// nitro code <https://github.com/Layr-Labs/nitro/blob/14f09745b74321f91d1f702c3e7bb5eb7d0e49ce/arbitrator/prover/src/kzgbn254.rs#L141>
+    /// could refactor in the future, such that both host and client can compute the proof
+    pub fn push_witness(&mut self, blob: &[u8], srs: &SRS) -> Result<(), KzgError> {
+        // TODO check if KZG can just take a reference to srs, it saves memory
+        let mut kzg = KZG::new(srs.clone());
+
+        let input = Blob::new(blob);
+        let input_poly = input.to_polynomial_eval_form();
+
+        kzg.calculate_and_store_roots_of_unity(blob.len() as u64)
+            .unwrap();
+
+        let mut commitment_bytes = vec![0u8; 0];
+
+        let commitment = kzg.commit_eval_form(&input_poly)?;
+
+        // TODO the library should have returned the bytes, or provide a helper
+        // for conversion. For both proof and commitment
+        let commitment_x_bigint: BigUint = commitment.x.into();
+        let commitment_y_bigint: BigUint = commitment.y.into();
+
+        append_left_padded_biguint_be(&mut commitment_bytes, &commitment_x_bigint);
+        append_left_padded_biguint_be(&mut commitment_bytes, &commitment_y_bigint);
+
+        let mut proof_bytes = vec![0u8; 0];
+
+        let proof = kzg.compute_blob_proof(&input, &commitment)?;
+        let proof_x_bigint: BigUint = proof.x.into();
+        let proof_y_bigint: BigUint = proof.y.into();
+
+        append_left_padded_biguint_be(&mut proof_bytes, &proof_x_bigint);
+        append_left_padded_biguint_be(&mut proof_bytes, &proof_y_bigint);
+
+        // push data into witness
+        self.write(
+            Bytes::copy_from_slice(blob),
+            Bytes::copy_from_slice(&commitment_bytes),
+            proof_bytes.into(),
+        );
+
+        Ok(())
+    }
+
+    /// helpful function to populate struct all together
+    pub fn write(&mut self, blob: Bytes, commitment: Bytes, proof: Bytes) {
+        self.eigenda_blobs.push(blob);
+        self.commitments.push(commitment);
+        self.proofs.push(proof);
+        info!("added a blob");
+    }
+
+    /// verify all the blobs contained within the struct. It ensures the integrity of the
+    /// blob with respect the commitment. Using the batch verification method is the reduction
+    /// the size of MSM. For n blob of 4096 field element, instead of n MSM of 4096 points,
+    /// it would only require O(1) MSM of n points.
+    pub fn batch_verify(&self) -> bool {
```

Comment:
```suggestion
    /// Verify all the blobs contained within the struct. It ensures the integrity of the
    /// blobs with respect to their commitments. Using batch verification instead of verifying each element one by one reduces
    /// the size of the MSM. For n blob of 4096 field element, instead of n MSM of 4096 points,
    /// only O(1) MSM of n points is required.
    pub fn batch_verify(&self) -> bool {
```